bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/channelhelper/HomeEnergyChannelHelper.java

```diff
@@ -42,11 +56,84 @@ public HomeEnergyChannelHelper(Thing thing, TimeZoneProvider timeZoneProvider) {
-                ? toQuantityType(localThing.getThermSetpointDefaultDuration(), Units.MINUTE)
-                : CHANNEL_PLANNING.equals(channelId)
-                        ? toStringType(localThing.getThermSchedules().stream().filter(schedule -> schedule.isSelected())
-                                .findFirst().map(NAThermProgram::getId).orElse(null))
-                        : null;
+        NAThermProgram currentProgram = localThing.getActiveProgram();
+        SetpointMode thermMode = localThing.getThermMode();
+
+        switch (channelId) {
+            case CHANNEL_SETPOINT_DURATION:
+                return toQuantityType(localThing.getThermSetpointDefaultDuration(), Units.MINUTE);
+            case CHANNEL_PLANNING:
+                return (currentProgram != null ? toStringType(currentProgram.getName()) : null);
+            case CHANNEL_SETPOINT_MODE:
+                if (thermMode != null) {
+                    switch (thermMode) {
+                        case PROGRAM:
+                        case HOME:
+                        case SCHEDULE:
+                            NATimeTableItem currentProgramMode = getCurrentProgramMode(localThing.getActiveProgram());
+                            if (currentProgram != null && currentProgramMode != null) {
+                                return new StringType(currentProgram
+                                        .getZone(String.valueOf(currentProgramMode.getZoneId())).getName());
+                            }
+                        case AWAY:
+                        case MANUAL:
+                        case FROST_GUARD:
+                            return new StringType(thermMode.name());
+                        case OFF:
+                        case MAX:
+                        case UNKNOWN:
+                            return UnDefType.UNDEF;
+                    }
+                }
+                return null;
+            case CHANNEL_SETPOINT_END_TIME:
+                if (thermMode != null) {
+                    switch (thermMode) {
+                        case PROGRAM:
+                        case HOME:
+                        case SCHEDULE:
+                            return toDateTimeType(getNextProgramTime(localThing.getActiveProgram()), zoneId);
+                        case AWAY:
+                        case MANUAL:
+                        case FROST_GUARD:
+                            return (localThing.getThermModeEndTime() > 0
+                                    ? toDateTimeType(localThing.getThermModeEndTime(), zoneId)
+                                    : UnDefType.UNDEF);
+                        case OFF:
+                        case MAX:
+                        case UNKNOWN:
+                            return UnDefType.UNDEF;
+                    }
+                }
+                return null;
+        }
+        return null;
+    }
+
+    private @Nullable NATimeTableItem getCurrentProgramMode(@Nullable NAThermProgram activeProgram) {
```

Comment:
If, wich I believe, this code is shared with Therm1SetpointChannelHelper, it could be moved to a common ancestor of both classes.